packages/core/src/credentials-provider/DefaultCredentialsProvider.spec.ts

```diff
@@ -0,0 +1,31 @@
+import { CredentialProvider } from './CredentialsProvider';
+import { EnvCredentialProvider } from './DefaultCredentialsProvider';
+import { reset, spy, when } from 'ts-mockito';
+
+describe('EnvCredentialProvider', () => {
+  describe('get', () => {
+    const spiedEnv = spy(process.env);
+    let provider!: CredentialProvider;
+
+    beforeEach(() => {
+      provider = new EnvCredentialProvider();
+    });
+
+    afterEach(() => reset(spiedEnv));
+
+    it('should return undefined if credentials in not provided', async () => {
+      const token = (await provider.get())?.token;
+
+      expect(token).toBeUndefined();
```

Comment:
Assert on the entire response, not on every field
```suggestion
      const result = await provider.get();

      expect(result).toBeUndefined();
```